File: PE3.py
###     Problem 3:  Largest prime factor of 600851475143
import math
def primefactors(number):
    og = number
    factors = []
    factorsMult = 1

    i = 2
    while(factorsMult!=og):
        while(number % i == 0):
            factors.append(i)
            number = number / i
        i = i + 1

        factorsMult = 1
        for factor in factors:
            factorsMult = factorsMult * factor
        
    return max(factors)

def isprime(num):
    for i in range(2, int(math.sqrt(num))):
        if(num % i == 0):
            return False
    return True

print(primefactors(600851475143))
###     Answer: 6857

File: PE3/PE3.cpp
#include<stdio.h>

int largest_prime_factor(long n) {
	int largest = 1;
	int i = 2;
	while( n!= 1 ) {
		while( n % i ==0 ) {
			if( i > largest ) {
				largest = i;
			}
			n = n / i;
		}
		i += 1;
	}
	return largest;
}

int main() {
	printf("%d\n", largest_prime_factor(600851475143));
	return 0;
}

// answer = 6857
